src/utils/rbac.ts

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { k8sCreateResource, K8sModelCommon, K8sVerb } from '@openshift/dynamic-plugin-sdk-utils';
+import memoize from 'lodash/memoize';
+import { SelfSubjectAccessReviewModel } from '../models/rbac';
+import { AccessReviewResourceAttributes, SelfSubjectAccessReviewKind } from '../types/rbac';
+import { useWorkspaceInfo } from './workspace-context-utils';
+
+export const checkAccess = memoize(async (group, resource, subresource, namespace, verb) => {
```

Comment:
noticed an issue with `SelfSubjectAccessReview` network calls. In application details page there has to be four different access review calls for the resources like Applications, environments, components, integrationTest but only one call (components) is made and that result is passed back to all other access review caller functions.
 
 Application details page access network call:
 
<img width="633" alt="Screenshot 2023-03-21 at 1 11 01 PM" src="https://user-images.githubusercontent.com/9964343/226548734-2c923478-08f9-41d5-b2a8-0de0136ff78b.png">


The issue is with memoize function in the checkAcess method, by default lodash will use the first argument passed as a cache key in this case the first argument is `group` .. since all four resources I mentioned is using the same group `appstudio.redhat.com` so all of them are sharing the same result. A better approach would be pass a second argument to the memoize function to resolve the cache key. something like `<group>-<resource>-<verb>` to be more precise.

Reply:
Thanks, did not realize this. Updated.